Submodels/MTSOL_call.py

```
@@ -198,34 +303,77 @@ def GenerateProcess(self,
+        if hasattr(self, "shutdown_event"):
+            self.shutdown_event.clear()
+        else:
+            self.shutdown_event = threading.Event()
+
+        # Stop any orphaned reader threads if they exist before starting the new subprocess
+        if getattr(self, "reader", None) and self.reader.is_alive():
+            # Signal the thread to stop
+            self.shutdown_event.set()
+
+            # Close stdout to unblock a blocking readline() call
+            if getattr(self, "process", None) and self.process.stdout:
```

Pylint (3.3.7)

[error] Access to member 'process' before its definition line 339

(E0203)